Fucntional/#4756-create-new-endpoint-to-return-list-of-active-locations

JJJazl opened the pull request:
1) added new endpoint for get all active locations in SuperAdminController
2) added method getActiveLocations() in SuperAdminService
3) added method getActiveLocations() int SuperAdminServiceImpl
4) added test method getActiveLocations() in SuperAdminControllerTest
5) added test method getActiveLocationsTest in SuperAdminServiceImplTest
6) changed SecurityConfig

renat-safarov committed:
1) added new endpoint for get all active locations in SuperAdminController
2) added method getActiveLocations() in SuperAdminService
3) added method getActiveLocations() int SuperAdminServiceImpl
4) added test method getActiveLocations() in SuperAdminControllerTest
5) added test method getActiveLocationsTest in SuperAdminServiceImplTest
6) changed SecurityConfig

RomaMocherniuk commented on core/src/main/java/greencity/configuration/SecurityConfig.java:
```diff
@@ -91,6 +91,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 ADMIN_LINK + "/notification/get-template/{id}",
                 SUPER_ADMIN_LINK + "/get-all-receiving-station",
                 SUPER_ADMIN_LINK + "/getLocations",
+                SUPER_ADMIN_LINK + "//getActiveLocations",
```
please remove extra `/`

RomaMocherniuk commented on service/src/test/java/greencity/service/ubs/SuperAdminServiceImplTest.java:
```diff
@@ -290,6 +290,17 @@ void getAllLocationTest() {
+
+        when(regionRepository.findAll()).thenReturn(regionList);
+
+        superAdminService.getActiveLocations();
```
please check response too

RomaMocherniuk replied:
please add a test if `regionRepository.findAll()` returns inactive locations too to check this line of code https://github.com/ita-social-projects/GreenCityUBS/pull/922/files#diff-749434d8995cf471e2c87f8ee3afdf70c2cc5301040766ab9c3b60d2703ae0bfR248

renat-safarov committed:
1) added method findRegionsWithActiveLocations() in RegionRepository
2) removed 1 extra / in SecurityConfig
3) changed description of new method getActiveLocations() in SuperAdminController
4) changed method getActiveLocations() int SuperAdminServiceImpl

renat-safarov committed:
1) formatted method getActiveLocations() in SuperAdminServiceImpl

renat-safarov committed:
1) changed test method getActiveLocationsTest() in SuperAdminServiceImplTest
